actionview/lib/action_view/cache_expiry.rb

```diff
@@ -40,7 +40,7 @@ def build_watcher
 
             if @watched_dirs != dirs_to_watch
               @watched_dirs = dirs_to_watch
-              new_watcher = @watcher_class.new([], @watched_dirs) do
+              new_watcher = @watcher_class.new([], @watched_dirs, filter_path: Rails.root.to_s) do
```

Comment:
This is coupling Action View with Railties in the wrong place.

We should pass a configuration to ActionView in its Railtie and use it here.

Reply:
agree, rewrote the code, also added `filter_path` to `EventedFileUpdateChecker`, and added tests